### PR TITLE
build: don't depend on libidn2

### PR DESCRIPTION
**Pull request by Totktonada**
The recent libcurl update in commit 0919f390802f ("third_party: update libcurl from 8.8.0 to 8.10.1") reveals an old problem in our libcurl's build script: it attempts to disable the libidn2 dependency using the `-DHAVE_LIBIDN2=OFF` option instead of `-DUSE_LIBIDN2=OFF`.

It doesn't work after a recent curl building machinery change, see https://github.com/curl/curl/pull/14555

This commit fixes the problem by using `-DUSE_LIBIDN2=OFF`.

Reported in #10621

### CODE REVIEW

File: cmake/BuildLibCURL.cmake [hunk 1]
```diff
@@ -150,8 +150,7 @@ macro(curl_build)
     list(APPEND LIBCURL_CMAKE_FLAGS "-DCURL_USE_RUSTLS=OFF")
     list(APPEND LIBCURL_CMAKE_FLAGS "-DCURL_USE_WOLFSSH=OFF")
     list(APPEND LIBCURL_CMAKE_FLAGS "-DCURL_USE_GSASL=OFF")
-    list(APPEND LIBCURL_CMAKE_FLAGS "-DHAVE_LIBIDN2=OFF")
```
Comment (ligurio) [inline cmake/BuildLibCURL.cmake:153]: Should we put `DHAVE_LIBIDN2` to `cmake/curl-excluded-options.txt`?

**Review by ligurio — APPROVED** [review]
Thanks for the patch! LGTM with a minor comment.

File: cmake/BuildLibCURL.cmake (hunk 1 above)
Comment (Totktonada) [reply cmake/BuildLibCURL.cmake:153]: I've verified that it is not in the `cmake -LHA` output, at least on curl `8.10.1`. It seems, now there is no point to place it neither to the passed options, nor to the ignored ones.

Comment (Col-Waltz) [reply cmake/BuildLibCURL.cmake:153]: I agree with this changes, `USE_LIBIDN2` was set to ON by default in `curl/CMakeLists.txt` so I didn't know that we don't depend on it.

**Review by Col-Waltz — APPROVED** [review]
Thanks for the patch!

I agree with this changes, `USE_LIBIDN2` was set to ON by default in `curl/CMakeLists.txt` so I didn't know that we don't depend on it.

File: cmake/BuildLibCURL.cmake (hunk 1 above)
Comment (ligurio) [reply cmake/BuildLibCURL.cmake:153]: Thanks for explanation!

**Comment by Totktonada** [comment]
Backported to `release/3.2` in PR #10676 and to `release/2.11` in PR #10677.

**Pull request merged**